### PR TITLE
[file_selector_platform_interface] File selector nnbd

### PR DESCRIPTION
**Pull request by SirusCodes**
Migrating `file_selector_interface` to null-safety

*List which issues are fixed by this PR. You must list at least one issue.*
https://github.com/flutter/flutter/issues/75235

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: packages/file_selector/file_selector_platform_interface/lib/src/platform_interface/file_selector_interface.dart [hunk 1]
```diff
@@ -38,37 +38,37 @@ abstract class FileSelectorPlatform extends PlatformInterface {
   }
 
   /// Open file dialog for loading files and return a file path
```
Comment (stuartmorgan) [inline packages/file_selector/file_selector_platform_interface/lib/src/platform_interface/file_selector_interface.dart:40]: I think it's worth adding a comment that it returns null if the user cancel's the operation, to make it clear why it's nullable. (Same for `getSavePath` and `getDirectoryPath`.)

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
A couple of comments even though this is still marked as a Draft.

Most importantly though, this can't be a single PR. You need to do a separate PR for each package, and they need to be landed and published in order, or it won't build. That's why the steps are listed separately in the issue.

**Comment by SirusCodes** [comment]
> Most importantly though, this can't be a single PR. You need to do a separate PR for each package, and they need to be landed and published in order, or it won't build. That's why the steps are listed separately in the issue.

Ohh I thought I could make a single PR😅

File: packages/file_selector/file_selector_platform_interface/lib/src/platform_interface/file_selector_interface.dart (hunk 1 above)
Comment (SirusCodes) [reply packages/file_selector/file_selector_platform_interface/lib/src/platform_interface/file_selector_interface.dart:40]: Yaa sounds logical..

**Comment by SirusCodes** [comment]
@stuartmorgan `2.0.0-nullsafety.0` is good or `1.1.0-nullsafety.0`? for versioning as per [migration guide](https://dart.dev/null-safety/migration-guide#package-version) I feel both are right can you please let me know which one is better?
Previous was `1.0.3+1`

**Comment by stuartmorgan** [comment]
> @stuartmorgan `2.0.0-nullsafety.0` is good or `1.1.0-nullsafety.0`? for versioning as per [migration guide](https://dart.dev/null-safety/migration-guide#package-version) I feel both are right

As the guide you linked to says, bumping the major version is required in this case.

**Comment by SirusCodes** [comment]
> > @stuartmorgan `2.0.0-nullsafety.0` is good or `1.1.0-nullsafety.0`? for versioning as per [migration guide](https://dart.dev/null-safety/migration-guide#package-version) I feel both are right
> 
> As the guide you linked to says, bumping the major version is required in this case.

So `2.0.0-nullsafety.0` is fine?

File: packages/file_selector/file_selector_platform_interface/pubspec.yaml [hunk 2]
```diff
@@ -3,23 +3,23 @@ description: A common platform interface for the file_selector plugin.
...
 
 dependencies:
   flutter:
     sdk: flutter
   meta: ^1.0.5
-  http: ^0.12.0+1
-  plugin_platform_interface: ^1.0.1
-  cross_file: ^0.2.0
+  http: ^0.13.0-nullsafety.0
+  plugin_platform_interface: 1.1.0-nullsafety.2
```
Comment (stuartmorgan) [inline packages/file_selector/file_selector_platform_interface/pubspec.yaml:13]: This needs a `^` (see the error message on the failing `publishable` CI bot)

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Looks good overall, thanks! Just a couple notes to address the bot failures:

You'll need to update https://github.com/flutter/plugins/blob/master/script/nnbd_plugins.sh to list this an an NNBD plugin so that it isn't built by the automated tests that run on stable, to fix one of the failing CI bots.

File: packages/file_selector/file_selector_platform_interface/pubspec.yaml (hunk 2 above)
Comment (SirusCodes) [reply packages/file_selector/file_selector_platform_interface/pubspec.yaml:13]: Yes will do that. Thanks, can expect commit in 15mins

**Review by stuartmorgan — APPROVED** [review]
Looks good, thanks! I'll get this landed and published soon.

**Comment by SirusCodes** [comment]
Thanks, will make next PR after this is published ;)

**Comment by stuartmorgan** [comment]
Published.

File: packages/file_selector/file_selector_web/pubspec.yaml [hunk 3]
```diff
@@ -22,8 +22,8 @@ dependencies:
 dev_dependencies:
   flutter_test:
     sdk: flutter
-  mockito: ^4.1.1
-  pedantic: ^1.8.0
+  mockito: ^5.0.0-nullsafety.5
+  pedantic: ^1.10.0-nullsafety.3
```
Comment (ditman) [inline packages/file_selector/file_selector_web/pubspec.yaml:26]: I think this shouldn't have been pushed in this change. Apologies, I just noticed!

Comment (stuartmorgan) [reply packages/file_selector/file_selector_web/pubspec.yaml:26]: Oof, not sure how I missed that. Want to push a quick revert on of this?

Comment (ditman) [reply packages/file_selector/file_selector_web/pubspec.yaml:26]: Here's the revert: https://github.com/flutter/plugins/pull/3528

Comment (SirusCodes) [reply packages/file_selector/file_selector_web/pubspec.yaml:26]: I'm sorry my bad 🥺

Comment (ditman) [reply packages/file_selector/file_selector_web/pubspec.yaml:26]: @SirusCodes not at all, I should have published my comment wayyyyy earlier :)

**Pull request merged**